File: java/src/test/java/org/hello/HelloAppTest.java
package org.hello;

import org.junit.jupiter.api.Test;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

class HelloAppTest {

  @Test
  void shouldPrintHelloMessage() {
    var messageText = "Hello Test!";
    var message = mock(HelloMessage.class);
    when(message.getText()).thenReturn(messageText);

    var console = mock(HelloConsole.class);

    var app = new HelloApp(message, console);
    app.printHello();

    verify(console).print(messageText);
  }
}

File: dotnet/Hello.Test/HelloAppTest.cs
using Hello.Main;
using Moq;

namespace Hello.Test;

public class HelloAppTest
{
    [Test]
    public void ShouldPrintHelloMessage()
    {
        var messageText = "Hello Test!";
        var messageMock = new Mock<HelloMessage>();
        messageMock.Setup(message => message.Text).Returns(messageText);
        var message = messageMock.Object;

        var consoleMock = new Mock<HelloConsole>();
        var console = consoleMock.Object;

        var app = new HelloApp(message, console);
        app.PrintHello();

        consoleMock.Verify(console => console.Print(messageText), Times.Once);
    }
}